js_modules/dagit/packages/core/src/ui/SectionedLeftNav.tsx

```diff
@@ -208,47 +213,49 @@ export const Section: React.FC<SectionProps> = (props) => {
       {visibleItems({type: 'asset-group', items: assetGroupItems})}
     </Box>
   );
-};
+});
 
-type PathMatch =
-  | {
-      repoPath: string;
-      pipelinePath: string;
-    }
-  | {
-      repoPath: string;
-      groupName: string;
-    };
+type PathMatch = {
+  repoPath: string;
+  pipelinePath?: string;
+  groupName?: string;
+};
 
 const usePathMatch = () => {
   const match = useRouteMatch<PathMatch>([
     '/workspace/:repoPath/(jobs|pipelines)/:pipelinePath',
     '/workspace/:repoPath/asset-groups/:groupName',
   ]);
-  if (!match) {
-    return null;
-  }
-  const repoAddress = repoAddressFromPath(match.params.repoPath);
-  if (!repoAddress) {
-    return null;
-  }
+  const {groupName, repoPath, pipelinePath} = match?.params || {};
 
-  return 'pipelinePath' in match.params
-    ? {
-        repoAddress,
-        itemName: explorerPathFromString(match.params.pipelinePath).pipelineName,
-        itemType: 'job' as const,
-      }
-    : {
-        repoAddress,
-        itemName: match.params.groupName,
-        itemType: 'asset-group' as const,
-      };
+  return React.useMemo(() => {
```

Comment:
Adding this useMemo was the reason for the changes here. Unfortunately the `match` object changes even if all three items inside it are identical strings, so the useMemo couldn't use `match` as it's dep.